In [1]:
from scipy.stats import norm, t
import numpy as np

<hr style="color: #009933; border: solid 1px">
<span style="color: #009933;">unknown std, $s$</span>

# <span style="color: #2455C3;">Keyboards</span>

A researcher is interesed in the effects of two different keyboard configurations (Qwerty and alphabetical) in cellphones on errors made in typiing text messages. To study this 25 participants ($n=25$) used each keyboard type, they typed a standarized 20 word text message within 30 seconds. The number of erros for each person using each type of keyboard was recorded. These participants where randomly assign to which type they used first. The data about errors made by each person is this:

##  <span style="color: #85100F;">Solution</span>

#### <span style="color: #85100F;">1. State the hypothesis</span>

$H_0$: there is no difference in the number of errors made in typing with any of the two diffrent keybords

$$\begin{array}{cc}
H_{0}: & \mu_{1}-\mu_{2}=0\\
H_{1}: & \mu_{1}-\mu_{2}\neq0
\end{array}\Rightarrow\begin{array}{cc}
H_{0}: & \mu_{D}=0\\
H_{1}: & \mu_{D}\neq0
\end{array}$$

#### <span style="color: #85100F;">2. Analyzing sample data</span>

In [3]:
# known data
# dependent two-sample test
n = len(qwerty)
mu = 0
alpha = 0.05

In [6]:
# samples - number of errors while typing
qwerty = np.array([6,6,2,7,8,8,2,3,5,7,10,5,4,7,5,7,4,5,2,5,3,4,4,4,4])
alphab = np.array([6,11,8,5,11,8,10,7,4,3,7,6,10,10,6,5,10,11,13,8,5,11,7,8,5])
diff = qwerty - alphab

In [10]:
# mean and s
print "xbarQ={:0.2f}, xbarA={:0.2f}".format(qwerty.mean(), alphab.mean())
xbarD = diff.mean()
s_D = diff.std(ddof=1)
SE = s_D / np.sqrt(n)
print "xbar_D={:0.2f}, s_D={:0.2f}, SE={:0.2f}".format(xbarD,
                                                        s_D, SE)

xbarQ=5.08, xbarA=7.80
xbar_D=-2.72, s_D=3.69, SE=0.74


#### <span style="color: #85100F;">3. Test statistic calculation</span>

In [11]:
t_score = (xbarD - mu) / SE
t_score

-3.684207583536927

#### <span style="color: #85100F;">4. Critical point determination</span>

In [12]:
t_critical = t.ppf(1-alpha/2, n-1)
t_critical

2.0638985616280205

#### <span style="color: #85100F;">5. Results interpretation</span>

<span style="color: #009933;">The <b>null hypothesis is rejected with $p<0.01$</b>, which means that <b>there is a significant difference in the number of errors made with each keyboard</b>. In fact, given that $d = -0.74$ the <b>number of errors using qwerty keyboard is significantly less, between 1.2 and 4.2 less errors</b> than using alphabetical keyboard.</span>

1) <u>Descriptive statistics:</u>
    $$\begin{array}{ccc}
    \mu & = & 0\\
    \bar{x}_D & = & -2.72\\
    s_D & = & 3.69
    \end{array}$$

2) <u>Inferential statistics:</u>

t(24)=-3.68, p=.001, two-tailed<br>
Confidence interval on the mean difference on number of errors<br>
95% CI = (-4.24 to -1.2)

3) <u>Effect size measures:</u>

* Cohen's d = -0.74
* R^2 = .36

So the <b>36%</b> of the <b>difference in erros while typing</b> for the sample of 25 peaple are <b>due to</b> the <b>keyboard type</b>.

We reject the null hypothesis with $p < 0.05$. In conclusion, given that $t-score = -3.68$, the use of alphabetical keyboard leads to significantly more errors in typing than the qwerty keyboard.

In [16]:
# P-value
P_value = 2*t.cdf(t_score, n-1)
print "P-value = {:0.3f}".format(P_value)

P-value = 0.001


In [18]:
# CI
t_char = t.ppf(1-alpha/2, n-1)
me = t_char * SE
print "95% CI = ({:0.2f} to {:0.2f})".format(xbarD-me, xbarD+me)

95% CI = (-4.24 to -1.20)


In [19]:
# Cohen's d
d = (xbarD - mu) / s_D
print "d = {:0.2f}".format(d)

d = -0.74


In [20]:
# R^2
r_squared = t_score**2 / (t_score**2 + (n-1))
print "R^2 = {:0.2f}".format(r_squared)

R^2 = 0.36
